src/model/types/account_id.rs

```diff
@@ -74,11 +75,31 @@ impl<'a> Deserialize<'a> for AccountId {
     }
 }
 
-fn is_eip155_account(account_id: &str) -> bool {
+#[derive(Debug, PartialEq, Eq, Error)]
+pub enum Eip155Error {
+    #[error("does not match regex")]
+    Regex,
+
+    #[error("does not pass ERC-55 checksum")]
+    Erc55,
+}
+
+fn is_valid_eip155_account(account_id: &str) -> Result<(), Eip155Error> {
     static PATTERN_CELL: OnceCell<regex::Regex> = OnceCell::new();
-    let pattern =
-        PATTERN_CELL.get_or_init(|| regex::Regex::new(r"^eip155:\d+:0x[0-9a-fA-F]{40}$").unwrap());
-    pattern.is_match(account_id)
+    let pattern = PATTERN_CELL
+        .get_or_init(|| regex::Regex::new(r"^eip155:\d+:0x(?<address>[0-9a-fA-F]{40})$").unwrap());
+
+    if let Some(caps) = pattern.captures(account_id) {
+        let address = &caps["address"];
```

Comment:
I'd do an `if let Some` here

Reply:
What's the error handling behavior if `address` isn't present? This is like a coding failure here, not a runtime error. If the regex matches then `address` should exist which is why I did `[]` instead of `.get()`

Reply:
`Index<&str>` only [panics if there isn't a capture group with the given name](https://docs.rs/regex/latest/regex/struct.Captures.html#impl-Index%3C%26str%3E-for-Captures%3C'h%3E).

But this could panic if I made a mistake in the regex such as conditionally matching. Looks like there's a slightly better alternative, extract, which panics [if the number of possible matching groups in this Captures value is not fixed to N in all circumstances](https://docs.rs/regex/latest/regex/struct.Captures.html#panics). So although it doesn't support named capture groups, it would result in catching any introduction of `|` operators in the regex. I switched to this in 1658283cce7913d315d6e78a5777e628aabf4031

Reply:
> What's the error handling behavior if `address` isn't present? This is like a coding failure here, not a runtime error. If the regex matches then `address` must exist which is why I did `[]` instead of `.get()`

Initially I'd just log the error and figure it out if/when it becomes a problem but you found a better approach now 👍